### PR TITLE
Add `sshLogins` to nodes endpoint on webapi (`GET /nodes`)

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR adds a field called `SSHLogins` to the `Server` struct that is returned by the endpoint on `/webapi/sites/:site/nodes`.

The reason behind this is to show in the web UI only the logins the user can actually use to connect to that node, instead of showing every login the user has access to as it is done currently.

See #6822 for more details.

Also, removed the field `sshLogins` from the user context endpoint (`GET /webapi/sites/:site/context`) since it is no longer necessary.

The frontend part of this implementation can be found in [this pr](https://github.com/gravitational/webapps/pull/873).

### CODE REVIEW

File: lib/web/ui/server.go [hunk 1]
```diff
@@ -88,19 +93,55 @@ func MakeServers(clusterName string, servers []types.Server) []Server {
...
+// getServerLogins returns the list of logins the role set has that is associated with this node
+func getServerLogins(server types.Server, roleSet services.RoleSet) []string {
+	allowed := []string{}
+	denied := []string{}
+	for _, role := range roleSet {
+		// remove any denied logins, even if they don't apply to that node's labels
+		denied = append(denied, role.GetLogins(types.Deny)...)
+
+		// add only logins related to that server
+		isAllowed, _, err := services.MatchLabels(role.GetNodeLabels(types.Allow), server.GetAllLabels())
```
Comment (Tener) [inline lib/web/ui/server.go:121]: This is reimplementing logic from `lib/services/role.go`. When working on related PR: https://github.com/gravitational/teleport/pull/10458, I made similar change and received feedback that these should go to the core of Teleport or be exposed via endpoint.

I went with the former and added changes here: https://github.com/gravitational/teleport/pull/10458/files#diff-9b3fca74b2b2d4ce46a466de333f6c3e7640c21da8bd587e1837aa7a113eb7e7

I think you can reimplement `getServerLogins` in a similar fashion to how `EnumerateDatabaseUsers` works, unless there are issues I cannot see now.

**Comment by ravicious** [comment]
> @ravicious hey, can you confirm Teleport Connect doesn't use this endpoint?
> AFAIK it only does gRPC calls but wanted to make sure since I'm not familiar with the project

Yeah, Connect uses `lib/client.TeleportClient` for pretty much all communication with the cluster. Connect is much closer to tsh in that regard.

I think the only web stuff we use is `/web/config.js`.

File: lib/web/ui/server.go (hunk 1 above)
Comment (mcbattirola) [reply lib/web/ui/server.go]: Thanks, that makes sense.

I moved the method to `lib/services/role.go` and implemented it like `EnumerateDatabaseUsers` on https://github.com/gravitational/teleport/pull/13266/commits/da94e407094841fdef1ca2902ae3902d9e639038.

I didn't find a good way to reuse the logic of `EnumerateDatabaseUsers`, even though the methods are mostly the same, so I ended up duplicating it.

We could abstract things like `role.GetDatabaseUsers / role.GetLogins` to a single method like `GetResourceUsers(resource string, type types.RoleConditionType)` but IDK if that would make sense, DB users and node logins aren't precisely the same thing.

I think it's better to duplicate the logic here than to make something overly complicated.
Please let me know what you think.

File: lib/services/role.go [hunk 2]
```diff
@@ -875,6 +875,51 @@ func (set RoleSet) EnumerateDatabaseUsers(database types.Database, extraUsers ..
...
+			}
+		}
+
+		result.wildcardDenied = result.wildcardDenied || wildcardDenied
+
+		if err := NewRoleSet(role).checkAccess(server, AccessMFAParams{Verified: true}); err == nil {
+			result.wildcardAllowed = result.wildcardAllowed || wildcardAllowed
+		}
+
+	}
```
[the remark this answers is not in the document]
Comment (Tener) [reply lib/services/role.go:910]: I agree. Wildcard handling was a big source of complexity.

File: lib/web/ui/server.go (hunk 1 above)
Comment (Tener) [reply lib/web/ui/server.go]: Thank you, this shapes up to be something I'd expect to see. 

I don't think the duplication is too bad as it is. For better or worse, we have a lot of duplication in that part of code already 🤷 . While repetitive in some cases, it gives us an easy way to be flexible in others: case in point with the wildcards treatment.

File: lib/services/role.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcbattirola) [reply lib/services/role.go]: Thanks for pointing that out, this makes things way simpler.

Added in https://github.com/gravitational/teleport/pull/13266/commits/2a57020797c761a6fcffb5a6a8354f3b0ca6d733. Also adds checks for denied logins on tests.

**Review by nklaassen — APPROVED** [review]

**Review by Tener — APPROVED** [review]

**Pull request merged**